language-support/hs/bindings/src/DA/Ledger/Types.hs

```diff
@@ -187,17 +188,23 @@ data Value
     | VContract ContractId
     | VList [Value]
     | VInt Int
-    | VDecimal Text -- TODO: Maybe use Haskell Decimal type
-    | VString Text
-    | VTimestamp MicroSecondsSinceEpoch
+    | VDecimal Text
```

Comment:
We're using `Fixed E10` as the payload in the DAML-LF AST. See https://github.com/digital-asset/daml/blob/705f014a888e2854e6776458abe4484b2929dbc2/compiler/daml-lf-ast/src/DA/Daml/LF/Ast/Base.hs#L195 and https://www.stackage.org/haddock/lts-13.30/base-4.12.0.0/Data-Fixed.html

Reply:
ok, I'll try this...

Reply:
ok, changing to `Fixed E10` as you suggest
requires some changes to the raise/lower convert functions (as expected)
Good job I have round-tripping tests in place, or else I would have introduced a bug!